--- hardware/libhardware_legacy/wifi/Android.mk	2013-03-23 01:15:31.162232028 +0400
+++ home/jied83/Android.mk	2013-03-23 20:20:21.751721382 +0400
@@ -40,7 +40,11 @@
 LOCAL_CFLAGS += -DWIFI_EXT_MODULE_NAME=\"$(WIFI_EXT_MODULE_NAME)\"
 endif
 
-LOCAL_SRC_FILES += wifi/wifi.c
+ifeq ($(TARGET_CUSTOM_WIFI),)
+    LOCAL_SRC_FILES += wifi/wifi.c
+else
+    LOCAL_SRC_FILES += $(TARGET_CUSTOM_WIFI)
+endif
 
 ifeq ($(BOARD_HAVE_SAMSUNG_WIFI),true)
     LOCAL_CFLAGS += -DSAMSUNG_WIFI
